packages/ocaml-base-compiler/ocaml-base-compiler.4.12.0~alpha1/opam

```diff
@@ -1,36 +1,41 @@
 opam-version: "2.0"
-synopsis: "latest 4.12 development, with afl-fuzz instrumentation"
+synopsis: "First alpha release of OCaml 4.12.0"
 maintainer: "platform@lists.ocaml.org"
+authors: ["Xavier Leroy" "Damien Doligez" "Alain Frisch" "Jacques Garrigue" "Didier Rémy" "Jérôme Vouillon"]
+homepage: "https://ocaml.org"
+bug-reports: "https://github.com/ocaml/ocaml/issues"
+dev-repo: "git://github.com/ocaml/ocaml#4.12"
 depends: [
   "ocaml" {= "4.12.0" & post}
   "base-unix" {post}
   "base-bigarray" {post}
   "base-threads" {post}
-  "ocaml-beta"
+  "ocaml-beta" {opam-version < "2.1"}
 ]
```

Comment:
```suggestion
  "ocaml-option-vanilla"
]
```

Reply:
Note it's remained `ocaml-options-vanilla` (in plural) - so you depend on "vanilla options" (i.e. no options) rather than "the vanilla option" (which, alas, OCaml does not in fact have 🍦)

I think this is already achieved, though, because of the conflict-class, which prevents ocaml-base-compiler switching to ocaml-variants already, @AltGr?

Reply:
Oh, do we want to move the remaining `ocaml-options-*` to their singular counterparts before that? I think it's a bit confusing and potential source of errors to have both.

Apart from that, the request itself isn't a blocker, it's more for the sake of completeness. I don't care about it that much.

Reply:
Ah yes, indeed, in the case of `ocaml-base-compiler`, we have two options:
- Have `ocaml-base-compiler` be just an alias for `ocaml-variants.x+options` & `ocaml-options-vanilla`
- Have it stand-alone as before
- (indeed 3: replace `base-compiler` by the version with options, not needing the `variants.x+options` at all. Not right away though)

The two should be equivalent in what you will get in the resulting switch ; my first take was option 1 here, to avoid duplication of the build instruction, but after some thought we decided to be more conservative and keep the stand-alone definition: it's less disruptive to current users of `ocaml-base-compiler`.

@dra27 yes, `ocaml-variants` and `ocaml-base-compiler` are mutually exclusive as before, and all options require `variants` so this is not needed; I guess @kit-ty-kate  wanted the presence of `options-vanilla` to be a reliable marker that you are on the vanilla case; since the `ocaml-options-vanilla` package has already been made compatible with `ocaml-base-compiler`, there would be no harm in adding this dependency here to that purpose I guess.